autoload/javacomplete/server.vim

```diff
@@ -46,7 +46,7 @@ endfunction
 
 function! javacomplete#server#Stop()
 	if s:Poll()
-		JavacompltePy vim.command('!kill %d'%bridgeState.pid())
+		JavacompltePy vim.command('!kill %d'%(bridgeState.pid()+1))
```

Comment:
When `bridgeState.pid()` is called it returns the pid for the process which was first spawned. This process has the signature (on my machine) of

 ```/bin/sh -c java  kg.ash.javavi.Javavi -sources "/home/ubuntu/.vim/bundle/vim-javacomplete2/libs/javavi//src:" -base "/home/ubuntu/.cache/javacomplete2/" -compiler "javac" -D 39502```

As you can see this is the process which is actually starting the java server. So the server's PID will be 1 greater than that of the shell which spawned it. 

Sending a SIGINT to the shell leaves the server process orphaned (I'm really not sure why, I would have thought it would forward the signal to the server). 

When a SIGINT is instead sent to the Java server (`PID +1`) it terminates, and then the shell, having its child process's execution finished, can also terminate.